cmd/remediation-aggregator/main.go

```diff
@@ -0,0 +1,426 @@
+}
+
+func parseConfig(cmd *cobra.Command) *aggregatorConfig {
+	var conf aggregatorConfig
+	conf.Content = getValidStringArg(cmd, "content")
+	conf.ScanName = getValidStringArg(cmd, "scan")
+	conf.Namespace = getValidStringArg(cmd, "namespace")
+	return &conf
+}
+
+func getValidStringArg(cmd *cobra.Command, name string) string {
+	val, _ := cmd.Flags().GetString(name)
+	if val == "" {
+		fmt.Fprintf(os.Stderr, "The command line argument '%s' is mandatory.\n", name)
+		os.Exit(1)
+	}
+	return val
+}
+
+func createCrClient(config *rest.Config) (*complianceCrClient, error) {
+	scheme := runtime.NewScheme()
+
+	v1.AddToScheme(scheme)
+	mcfgv1.AddToScheme(scheme)
+
+	scheme.AddKnownTypes(complianceoperatorv1alpha1.SchemeGroupVersion,
+		&complianceoperatorv1alpha1.ComplianceRemediation{})
+	scheme.AddKnownTypes(complianceoperatorv1alpha1.SchemeGroupVersion,
+		&complianceoperatorv1alpha1.ComplianceScan{})
+	scheme.AddKnownTypes(complianceoperatorv1alpha1.SchemeGroupVersion,
+		&metav1.CreateOptions{})
+
+	client, err := runtimeclient.New(config, runtimeclient.Options{
+		Scheme: scheme,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &complianceCrClient{
+		client: client,
+		scheme: scheme,
+	}, nil
+}
+
+func getScanRoleLabel(nodeSelector map[string]string) string {
+	if nodeSelector == nil {
+		return ""
+	}
+
+	// FIXME: should we protect against multiple labels and return
+	// an empty string if there are multiple?
+	for k := range nodeSelector {
+		if strings.HasPrefix(k, nodeRolePrefix) {
+			return strings.TrimPrefix(k, nodeRolePrefix)
+		}
+	}
+
+	return ""
+}
+
+func getScanConfigMaps(clientset *kubernetes.Clientset, scan, namespace string) ([]v1.ConfigMap, error) {
+	var cMapList *v1.ConfigMapList
+	var err error
+
+	// Look for configMap with this scan label
+	err = wait.PollImmediate(5*time.Second, 10*time.Minute, func() (bool, error) {
+		listOpts := metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("compliance-scan=%s", scan),
+		}
+
+		cMapList, err = clientset.CoreV1().ConfigMaps(namespace).List(listOpts)
+		if err != nil {
+			return false, err
+		}
+
+		if len(cMapList.Items) == 0 {
+			fmt.Printf("Scan %s has no results yet\n", scan)
+			return false, nil
+		}
+		return true, nil
+	})
+
+	if err != nil {
+		fmt.Printf("Error waiting for CMs of scan %s: %v\n", scan, err)
+		return nil, err
+	}
+
+	fmt.Printf("Scan %s has %d results\n", scan, len(cMapList.Items))
+	return cMapList.Items, nil
+}
+
+func readCompressedData(compressed string) (string, error) {
+	decoded, err := base64.StdEncoding.DecodeString(compressed)
+	if err != nil {
+		return "", err
+	}
+
+	compressedReader := bytes.NewReader(decoded)
+	bzReader, err := bzip2.NewReader(compressedReader, &bzip2.ReaderConfig{})
+	if err != nil {
+		return "", err
+	}
+	defer bzReader.Close()
+
+	// FIXME: probably unsafe, see https://haisum.github.io/2017/09/11/golang-ioutil-readall/
+	b, err := ioutil.ReadAll(bzReader)
```

Comment:
Do we need to read this already though? Why not pass the reader itself and just use it in the parsing function? that way we don't load the results in memory until they're needed.

Reply:
you're right, fixed locally